src/code/z_demo.c

```diff
@@ -1232,684 +1404,776 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
     }
 
     return size;
 }
 
-// Command 0x7: ? (Related to camera positons)
-s32 Cutscene_Command_07(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 unused) {
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_SetCamEye(PlayState* play, CutsceneContext* csCtx, u8* script, u8 unused) {
+    CsCmdGeneric* cmd = (CsCmdGeneric*)script;
     s32 size;
-    Vec3f sp3C;
-    Vec3f sp30;
+    Vec3f at;
+    Vec3f eye;
     Camera* subCam;
-    f32 sp28;
-
-    cmd += 8;
-    size = 8;
-
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((D_8015FCC2 < cmdBase->startFrame) || (D_8015FCC2 >= 0xF000))) {
-        csCtx->unk_1B = 1;
-        csCtx->subCamEyePoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1A != 0) {
-            D_8015FCC2 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+    f32 roll;
+
+    script += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
+
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((gCamEyeAppliedFrame < cmd->startFrame) || (gCamEyeAppliedFrame >= 0xF000))) {
+        csCtx->camEyeReady = true;
+        csCtx->camEyePoints = (CutsceneCameraPoint*)script;
+
+        if (csCtx->camAtReady) {
+            gCamEyeAppliedFrame = cmd->startFrame;
+
+            if (gUseCutsceneCam) {
                 subCam = Play_GetCamera(play, csCtx->subCamId);
                 subCam->player = NULL;
+
                 Play_ChangeCameraStatus(play, CAM_ID_MAIN, CAM_STAT_WAIT);
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_FREE0);
-                sp28 = csCtx->subCamLookAtPoints->cameraRoll * 1.40625f;
-                Camera_SetViewParam(subCam, CAM_VIEW_ROLL, &sp28);
-                sp3C.x = csCtx->subCamLookAtPoints->pos.x;
-                sp3C.y = csCtx->subCamLookAtPoints->pos.y;
-                sp3C.z = csCtx->subCamLookAtPoints->pos.z;
-                sp30.x = csCtx->subCamEyePoints->pos.x;
-                sp30.y = csCtx->subCamEyePoints->pos.y;
-                sp30.z = csCtx->subCamEyePoints->pos.z;
-                Play_CameraSetAtEye(play, csCtx->subCamId, &sp3C, &sp30);
-                Play_CameraSetFov(play, csCtx->subCamId, csCtx->subCamEyePoints->viewAngle);
+
+                roll = csCtx->camAtPoints->cameraRoll * 1.40625f;
+                Camera_SetViewParam(subCam, CAM_VIEW_ROLL, &roll);
+
+                at.x = csCtx->camAtPoints[0].pos.x;
+                at.y = csCtx->camAtPoints[0].pos.y;
+                at.z = csCtx->camAtPoints[0].pos.z;
+
+                eye.x = csCtx->camEyePoints[0].pos.x;
+                eye.y = csCtx->camEyePoints[0].pos.y;
+                eye.z = csCtx->camEyePoints[0].pos.z;
+
+                Play_CameraSetAtEye(play, csCtx->subCamId, &at, &eye);
+                Play_CameraSetFov(play, csCtx->subCamId, csCtx->camEyePoints->viewAngle);
```

Comment:
```suggestion
                eye.z = csCtx->camEyePoints[0].pos.z;

                Play_CameraSetAtEye(play, csCtx->subCamId, &at, &eye);
                Play_CameraSetFov(play, csCtx->subCamId, csCtx->camEyePoints[0].viewAngle);
```
or since it turns out `csCtx->camEyePoints` is a single `CutsceneCameraPoint`, revert to `->` ? (parkour!)